### PR TITLE
feat(java): Set parent pom to apache

### PR DESCRIPTION
**Pull request by CalvinKirs**
### Reason for this PR

https://maven.apache.org/pom/asf/
Apache Software Foundation Parent POM
contains settings that are likely to be useful to any Apache project that is building and releasing code with Maven. By using this project as a parent, a project gets these settings.
**If we have specific version requirements for certain plugins, we need to explicitly declare the corresponding versions in the project. Otherwise, it will default to the version specified in the parent POM.**
### What changes are included in this PR?
- Add mailing list and License etc

### CODE REVIEW

File: maven-projects/pom.xml [hunk 1]
```diff
@@ -22,20 +22,58 @@
...
+<!--    <issueManagement>
+        <system>GitHub</system>
+        <url>https://github.com/apache/graphar/issues</url>
+    </issueManagement>
+    <scm>
+        <connection>scm:git:https://github.com/apache/graphar.git</connection>
+        <developerConnection>scm:git:https://github.com/apache/graphar.git</developerConnection>
+        <tag>HEAD</tag>
+        <url>https://github.com/apache/graphar</url>
+    </scm>-->
```
Comment (CalvinKirs) [inline maven-projects/pom.xml:70]: https://lists.apache.org/thread/1frk5f146pbzprj0d1ftf15o4z6v3ocs  Do we have any plans to drop the "incubator" prefix? There's been an email thread discussing this in the incubator.

**Review by lixueclaire — APPROVED** [review]
LGTM. Thanks for making this change!

File: maven-projects/pom.xml (hunk 1 above)
Comment (lixueclaire) [reply maven-projects/pom.xml:70]: > https://lists.apache.org/thread/1frk5f146pbzprj0d1ftf15o4z6v3ocs Do we have any plans to drop the "incubator" prefix? There's been an email thread discussing this in the incubator.

Thanks for bringing this information to us. Are there specified steps or guidelines that we should follow in response to this?

Comment (CalvinKirs) [reply maven-projects/pom.xml:70]: Typically, related to infrastructure needs to be handled by the INFRA team. If we're planning to drop the "incubator" prefix, we'll just need to submit a ticket to the INFRA JIRA.
https://issues.apache.org/jira/projects/INFRA/summary

Comment (lixueclaire) [reply maven-projects/pom.xml:70]: > Typically, related to infrastructure needs to be handled by the INFRA team. If we're planning to drop the "incubator" prefix, we'll just need to submit a ticket to the INFRA JIRA. https://issues.apache.org/jira/projects/INFRA/summary

Thanks! I'll consult with other members and make a decision.

**Review by acezen — APPROVED** [review]
LGTM, thanks~

**Pull request merged**